chips/atmega2560-hal/src/pwm.rs

```diff
@@ -0,0 +1,363 @@
+//! let mut timer0 = Timer0Pwm::new(dp.TC0, pwm::Prescaler::Prescale64);
+//!
+//! let mut pb7 = portb.pb7.into_output(&mut portb.ddr).into_pwm(&mut timer0);
+//!
+//! pb7.set_duty(128);
+//! pb7.enable();
+//! ```
+//!
+//! Here is an overview of pins and which timer they work with:
+//!
+//! | Pin | Conversion Method |
+//! | --- | --- |
+//! | `PB4` | `.into_pwm(&mut timer2)` |
+//! | `PB5` | `.into_pwm(&mut timer1)` |
+//! | `PB6` | `.into_pwm(&mut timer1)` |
+//! | `PB7` | `.into_pwm(&mut timer0)` |
+//! | `PE3` | `.into_pwm(&mut timer3)` |
+//! | `PE4` | `.into_pwm(&mut timer3)` |
+//! | `PE5` | `.into_pwm(&mut timer3)` |
+//! | `PG5` | `.into_pwm(&mut timer0)` |
+//! | `PH3` | `.into_pwm(&mut timer4)` |
+//! | `PH4` | `.into_pwm(&mut timer4)` |
+//! | `PH5` | `.into_pwm(&mut timer4)` |
+//! | `PH6` | `.into_pwm(&mut timer2)` |
+//! | `PL3` | `.into_pwm(&mut timer5)` |
+//! | `PL4` | `.into_pwm(&mut timer5)` |
+//! | `PL5` | `.into_pwm(&mut timer5)` |
+
+use crate::port::{portb, porte, portg, porth, portl};
+pub use avr_hal::pwm::*;
+
+avr_hal::impl_pwm! {
+    /// Use `TC0` for PWM (pins `PB7`, `PG5`)
+    ///
+    /// # Example
+    /// ```
+    /// let mut portb = dp.PORTB.split();
+    /// let mut portg = dp.PORTG.split();
+    /// let mut timer0 = Timer0Pwm::new(dp.TC0, pwm::Prescaler::Prescale64);
+    ///
+    /// let mut pb7 = portb.pb7.into_output(&mut portb.ddr).into_pwm(&mut timer0);
+    /// let mut pg5 = portg.pg5.into_output(&mut portg.ddr).into_pwm(&mut timer0);
+    ///
+    /// pb7.set_duty(128);
+    /// pb7.enable();
+    /// ```
+    pub struct Timer0Pwm {
+        timer: crate::atmega2560::TC0,
+        init: |tim, prescaler| {
+            tim.tccr0a.modify(|_, w| w.wgm0().pwm_fast());
+            tim.tccr0b.modify(|_, w| match prescaler {
+                Prescaler::Direct => w.cs0().direct(),
+                Prescaler::Prescale8 => w.cs0().prescale_8(),
+                Prescaler::Prescale64 => w.cs0().prescale_64(),
+                Prescaler::Prescale256 => w.cs0().prescale_256(),
+                Prescaler::Prescale1024 => w.cs0().prescale_1024(),
+            });
+        },
+        pins: {
+            portb::PB7: {
+                ocr: ocr0a,
+                into_pwm: |tim| if enable {
+                    tim.tccr0a.modify(|_, w| w.com0a().match_clear());
+                } else {
+                    tim.tccr0a.modify(|_, w| w.com0a().disconnected());
+                },
+            },
+            portg::PG5: {
+                ocr: ocr0b,
+                into_pwm: |tim| if enable {
+                    tim.tccr0a.modify(|_, w| w.com0b().match_clear());
+                } else {
+                    tim.tccr0a.modify(|_, w| w.com0b().disconnected());
+                },
+            },
+        },
+    }
+}
+
+avr_hal::impl_pwm! {
+    /// Use `TC1` for PWM (pins `PB5`, `PB6`)
+    ///
+    /// Note: `PB7` is not currently supported (enter in conflict with `PB7` from timer0).
```

Comment:
You can support one pin from multiple timers; take a look at `atmega32u4-hal` where this is done, for example.  You'll have one method `.into_pwm()` for the default timer that you think most people would want to use and a secondary `.into_pwm1()` for the alternative (or `.into_pwm(whatever timer number)()`).

Please also mention this in the overview table as is done in `atmega32u4-hal`.

Reply:
Thanks for the pointer, I'll have a look and submit an update